### PR TITLE
[Optional Reference] LPS-185552 Convert the Optional Dynamic Greedy references to static references using Snapshot in class ThreadLocalScopeContextScopeChecker

### PR DESCRIPTION
**Pull request by haydenzheng314**
Ticket: https://issues.liferay.com/browse/LPS-185552
Thanks

### CODE REVIEW

**Comment by LisztLiszt** [comment]
ci:test:startup

**Comment by LisztLiszt** [comment]
ci:test:oauth2

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/oauth2-provider/oauth2-provider-scope-impl/src/main/java/com/liferay/oauth2/provider/scope/internal/ThreadLocalScopeContextScopeChecker.java [hunk 1]
```diff
@@ -107,10 +98,7 @@ public boolean checkScope(String scope) {
 		}
 
 		Collection<OAuth2ScopeGrant> oAuth2ScopeGrants =
-			_oAuth2ScopeGrantLocalService.getOAuth2ScopeGrants(
-				_companyIdThreadLocal.get(), _applicationNameThreadLocal.get(),
-				_bundleSymbolicNameThreadLocal.get(),
-				_accessTokenThreadLocal.get());
+			_getOAuth2ScopeGrants();
```
Comment (dantewang) [inline modules/apps/oauth2-provider/oauth2-provider-scope-impl/src/main/java/com/liferay/oauth2/provider/scope/internal/ThreadLocalScopeContextScopeChecker.java:101]: @LisztLiszt After extracting the methods, please also check if you can inline the variable.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Pull request closed without merging**